code/modules/admin/admin_verbs.dm

```diff
@@ -387,6 +392,61 @@ GLOBAL_LIST_INIT(view_runtimes_verbs, list(
 		// TODO: SStgui.on_transfer() to move windows from old and new
 		SSblackbox.record_feedback("tally", "admin_verb", 1, "Aghost") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
 
+
+/client/proc/admin_observe(mob/target)
+	set category = "Admin"
+	set name = "Aobserve"
+
+	if(!check_rights(R_ADMIN|R_MOD|R_MENTOR))
+		return
+
+	var/actually_admin = check_rights(R_ADMIN|R_MOD)
+
+
+	// todo what happens if a user we're observing logs out?
+
+	if(isnewplayer(mob))
+		to_chat(src, "<span class='warning'>You cannot aobserve while in the lobby. Please join or observe first.</span>")
+		return
+
+
+	if(isnull(target))
+		// choose a target
+		target = tgui_input_list(mob, "Select a mob to observe", "Aobserve", GLOB.player_list)
+		if(isnull(target))
+			return
+
+	if(target == src)
+		to_chat(src, "<span class='warning'>You can't observe yourself!</span>")
+		return
+
+	if(isobserver(mob))
+		var/mob/dead/observer/ghost = mob
+		if(ghost.mob_observed)
+			// un-follow them
+			ghost.cleanup_observe()
+			if(!actually_admin)
+				REMOVE_TRAIT(mob.mind, TRAIT_MOBSERVE, UNIQUE_TRAIT_SOURCE(target))
+			log_admin("[key_name(src)] has de-activated Aobserve")
```

Comment:
This should probably happen BEFORE asking for a target to observe, so you don't have to pick someone before un-observing.

Reply:
Yup, that's happening in some changes I'll put up shortly.